In [1]:
import pandas as pd
import numpy as np

# Loading Datasets

## Daily Activity Dataset

In [2]:
dfAct = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv')
dfAct

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366    4/12/2016       13162       8.500000         8.500000   
1    1503960366    4/13/2016       10735       6.970000         6.970000   
2    1503960366    4/14/2016       10460       6.740000         6.740000   
3    1503960366    4/15/2016        9762       6.280000         6.280000   
4    1503960366    4/16/2016       12669       8.160000         8.160000   
..          ...          ...         ...            ...              ...   
935  8877689391     5/8/2016       10686       8.110000         8.110000   
936  8877689391     5/9/2016       20226      18.250000        18.250000   
937  8877689391    5/10/2016       10733       8.150000         8.150000   
938  8877689391    5/11/2016       21420      19.559999        19.559999   
939  8877689391    5/12/2016        8064       6.120000         6.120000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                1.88                      0.55   
1                         0.0                1.57                      0.69   
2                         0.0                2.44                      0.40   
3                         0.0                2.14                      1.26   
4                         0.0                2.71                      0.41   
..                        ...                 ...                       ...   
935                       0.0                1.08                      0.20   
936                       0.0               11.10                      0.80   
937                       0.0                1.35                      0.46   
938                       0.0               13.22                      0.41   
939                       0.0                1.82                      0.04   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   6.06                     0.00                 25   
1                   4.71                     0.00                 21   
2                   3.91                     0.00                 30   
3                   2.83                     0.00                 29   
4                   5.04                     0.00                 36   
..                   ...                      ...                ...   
935                 6.80                     0.00                 17   
936                 6.24                     0.05                 73   
937                 6.28                     0.00                 18   
938                 5.89                     0.00                 88   
939                 4.25                     0.00                 23   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                     13                   328               728      1985  
1                     19                   217               776      1797  
2                     11                   181              1218      1776  
3                     34                   209               726      1745  
4                     10                   221               773      1863  
..                   ...                   ...               ...       ...  
935                    4                   245              1174      2847  
936                   19                   217              1131      3710  
937                   11                   224              1187      2832  
938                   12                   213              1127      3832  
939                    1                   137               770      1849  

[940 rows x 15 columns]

In [3]:
list(dfAct.columns.values)

['Id',
 'ActivityDate',
 'TotalSteps',
 'TotalDistance',
 'TrackerDistance',
 'LoggedActivitiesDistance',
 'VeryActiveDistance',
 'ModeratelyActiveDistance',
 'LightActiveDistance',
 'SedentaryActiveDistance',
 'VeryActiveMinutes',
 'FairlyActiveMinutes',
 'LightlyActiveMinutes',
 'SedentaryMinutes',
 'Calories']

In [4]:
dfAct.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  
count   940.000000  
mean   2303.609574  
std     718.166862  
min       0.000000  
25%    1828.500000  
50%    2134.000000  
75%    2793.250000  
max    4900.000000

### Key Takeaways:

    * Average(Mean) Sedentry minutes is 991.2 minutes or 16.5 hours/day! That's almost close to 70% of a day.
    
    * Most users are lightly active users.
    
    * Average total steps per day is 7637. Recommended total steps is 10,000 according to CDC.


## Daily Calories Dataset

In [5]:
dfCal = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/dailyCalories_merged.csv')
dfCal

Id ActivityDay  Calories
0    1503960366   4/12/2016      1985
1    1503960366   4/13/2016      1797
2    1503960366   4/14/2016      1776
3    1503960366   4/15/2016      1745
4    1503960366   4/16/2016      1863
..          ...         ...       ...
935  8877689391    5/8/2016      2847
936  8877689391    5/9/2016      3710
937  8877689391   5/10/2016      2832
938  8877689391   5/11/2016      3832
939  8877689391   5/12/2016      1849

[940 rows x 3 columns]

In [6]:
dfCal.describe()

Id     Calories
count  9.400000e+02   940.000000
mean   4.855407e+09  2303.609574
std    2.424805e+09   718.166862
min    1.503960e+09     0.000000
25%    2.320127e+09  1828.500000
50%    4.445115e+09  2134.000000
75%    6.962181e+09  2793.250000
max    8.877689e+09  4900.000000

## Daily Intensity Dataset

In [7]:
dfInt = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/dailyIntensities_merged.csv')
dfInt

Id ActivityDay  SedentaryMinutes  LightlyActiveMinutes  \
0    1503960366   4/12/2016               728                   328   
1    1503960366   4/13/2016               776                   217   
2    1503960366   4/14/2016              1218                   181   
3    1503960366   4/15/2016               726                   209   
4    1503960366   4/16/2016               773                   221   
..          ...         ...               ...                   ...   
935  8877689391    5/8/2016              1174                   245   
936  8877689391    5/9/2016              1131                   217   
937  8877689391   5/10/2016              1187                   224   
938  8877689391   5/11/2016              1127                   213   
939  8877689391   5/12/2016               770                   137   

     FairlyActiveMinutes  VeryActiveMinutes  SedentaryActiveDistance  \
0                     13                 25                     0.00   
1                     19                 21                     0.00   
2                     11                 30                     0.00   
3                     34                 29                     0.00   
4                     10                 36                     0.00   
..                   ...                ...                      ...   
935                    4                 17                     0.00   
936                   19                 73                     0.05   
937                   11                 18                     0.00   
938                   12                 88                     0.00   
939                    1                 23                     0.00   

     LightActiveDistance  ModeratelyActiveDistance  VeryActiveDistance  
0                   6.06                      0.55                1.88  
1                   4.71                      0.69                1.57  
2                   3.91                      0.40                2.44  
3                   2.83                      1.26                2.14  
4                   5.04                      0.41                2.71  
..                   ...                       ...                 ...  
935                 6.80                      0.20                1.08  
936                 6.24                      0.80               11.10  
937                 6.28                      0.46                1.35  
938                 5.89                      0.41               13.22  
939                 4.25                      0.04                1.82  

[940 rows x 10 columns]

In [8]:
dfInt.describe()

Id  SedentaryMinutes  LightlyActiveMinutes  \
count  9.400000e+02        940.000000            940.000000   
mean   4.855407e+09        991.210638            192.812766   
std    2.424805e+09        301.267437            109.174700   
min    1.503960e+09          0.000000              0.000000   
25%    2.320127e+09        729.750000            127.000000   
50%    4.445115e+09       1057.500000            199.000000   
75%    6.962181e+09       1229.500000            264.000000   
max    8.877689e+09       1440.000000            518.000000   

       FairlyActiveMinutes  VeryActiveMinutes  SedentaryActiveDistance  \
count           940.000000         940.000000               940.000000   
mean             13.564894          21.164894                 0.001606   
std              19.987404          32.844803                 0.007346   
min               0.000000           0.000000                 0.000000   
25%               0.000000           0.000000                 0.000000   
50%               6.000000           4.000000                 0.000000   
75%              19.000000          32.000000                 0.000000   
max             143.000000         210.000000                 0.110000   

       LightActiveDistance  ModeratelyActiveDistance  VeryActiveDistance  
count           940.000000                940.000000          940.000000  
mean              3.340819                  0.567543            1.502681  
std               2.040655                  0.883580            2.658941  
min               0.000000                  0.000000            0.000000  
25%               1.945000                  0.000000            0.000000  
50%               3.365000                  0.240000            0.210000  
75%               4.782500                  0.800000            2.052500  
max              10.710000                  6.480000           21.920000

## Daily Steps Dataset

In [9]:
dfStep = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/dailysteps_merged.csv')
dfStep.head()

Id ActivityDay  StepTotal
0  1503960366   4/12/2016      13162
1  1503960366   4/13/2016      10735
2  1503960366   4/14/2016      10460
3  1503960366   4/15/2016       9762
4  1503960366   4/16/2016      12669

In [10]:
dfStep.describe()

Id     StepTotal
count  9.400000e+02    940.000000
mean   4.855407e+09   7637.910638
std    2.424805e+09   5087.150742
min    1.503960e+09      0.000000
25%    2.320127e+09   3789.750000
50%    4.445115e+09   7405.500000
75%    6.962181e+09  10727.000000
max    8.877689e+09  36019.000000

## Daily Sleep Dataset

In [11]:
dfSleep = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv')
dfSleep

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0    1503960366  4/12/2016 12:00:00 AM                  1                 327   
1    1503960366  4/13/2016 12:00:00 AM                  2                 384   
2    1503960366  4/15/2016 12:00:00 AM                  1                 412   
3    1503960366  4/16/2016 12:00:00 AM                  2                 340   
4    1503960366  4/17/2016 12:00:00 AM                  1                 700   
..          ...                    ...                ...                 ...   
408  8792009665  4/30/2016 12:00:00 AM                  1                 343   
409  8792009665   5/1/2016 12:00:00 AM                  1                 503   
410  8792009665   5/2/2016 12:00:00 AM                  1                 415   
411  8792009665   5/3/2016 12:00:00 AM                  1                 516   
412  8792009665   5/4/2016 12:00:00 AM                  1                 439   

     TotalTimeInBed  
0               346  
1               407  
2               442  
3               367  
4               712  
..              ...  
408             360  
409             527  
410             423  
411             545  
412             463  

[413 rows x 5 columns]

## Weight Log Dataset

In [12]:
dfw = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/weightLogInfo_merged.csv')
dfw

Id                   Date    WeightKg  WeightPounds   Fat  \
0   1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.0   
1   1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147   NaN   
2   1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120   NaN   
3   2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104   NaN   
4   2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875   NaN   
..         ...                    ...         ...           ...   ...   
62  8877689391    5/6/2016 6:43:35 AM   85.000000    187.392923   NaN   
63  8877689391    5/8/2016 7:35:53 AM   85.400002    188.274775   NaN   
64  8877689391    5/9/2016 6:39:44 AM   85.500000    188.495234   NaN   
65  8877689391   5/11/2016 6:51:47 AM   85.400002    188.274775   NaN   
66  8877689391   5/12/2016 6:42:53 AM   84.000000    185.188300   NaN   

          BMI  IsManualReport          LogId  
0   22.650000            True  1462233599000  
1   22.650000            True  1462319999000  
2   47.540001           False  1460509732000  
3   21.450001            True  1461283199000  
4   21.690001            True  1463097599000  
..        ...             ...            ...  
62  25.440001           False  1462517015000  
63  25.559999           False  1462692953000  
64  25.610001           False  1462775984000  
65  25.559999           False  1462949507000  
66  25.139999           False  1463035373000  

[67 rows x 8 columns]

## Hourly Steps Dataset

In [13]:
dfHStep = pd.read_csv('/Users/suryanshjamwal/Desktop/Analytics Case studies/Fitabase Data 4.12.16-5.12.16/hourlySteps_merged.csv')
dfHStep.head()

Id           ActivityHour  StepTotal
0  1503960366  4/12/2016 12:00:00 AM        373
1  1503960366   4/12/2016 1:00:00 AM        160
2  1503960366   4/12/2016 2:00:00 AM        151
3  1503960366   4/12/2016 3:00:00 AM          0
4  1503960366   4/12/2016 4:00:00 AM          0

#### Seems like the datasets have common attributes and values.
#### We only use 'dailyActivity_merged' as the main dataframe as 'dailyCalories_merged',
#### 'dailyIntensities_merged' and 'dailysteps_merged' are all subsets of it. 
#### We verified that in the sql file.

# Data cleaning and transformation

## Checking for number of distinct values

In [14]:
dfAct['Id'].nunique()

33

In [15]:
dfSleep['Id'].nunique()

24

In [16]:
dfw['Id'].nunique()

8

In [17]:
dfHStep['Id'].nunique()

33

#### The weight info database is limited to 8 unique ID's. It would hinder with providing any concrete conclusions.
#### Hence, it must be excluded from the analyis.

## Checking for duplicates

In [18]:
len(dfAct)-len(dfAct.drop_duplicates())

0

In [19]:
len(dfSleep)-len(dfSleep.drop_duplicates())

3

In [20]:
len(dfHStep)-len(dfHStep.drop_duplicates())

0

In [21]:
# dropping duplicates in Sleep dataset

dfS = dfSleep.drop_duplicates()
len(dfS)-len(dfS.drop_duplicates())

# dfS is the updated Sleep dataset

0

## Converting timestamps

In [22]:
dfAct['ActivityDate'] = pd.to_datetime(dfAct['ActivityDate'], format='%m/%d/%Y')
dfAct['day_of_week'] = dfAct['ActivityDate'].dt.day_name()

In [23]:
dfAct

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366   2016-04-12       13162       8.500000         8.500000   
1    1503960366   2016-04-13       10735       6.970000         6.970000   
2    1503960366   2016-04-14       10460       6.740000         6.740000   
3    1503960366   2016-04-15        9762       6.280000         6.280000   
4    1503960366   2016-04-16       12669       8.160000         8.160000   
..          ...          ...         ...            ...              ...   
935  8877689391   2016-05-08       10686       8.110000         8.110000   
936  8877689391   2016-05-09       20226      18.250000        18.250000   
937  8877689391   2016-05-10       10733       8.150000         8.150000   
938  8877689391   2016-05-11       21420      19.559999        19.559999   
939  8877689391   2016-05-12        8064       6.120000         6.120000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                1.88                      0.55   
1                         0.0                1.57                      0.69   
2                         0.0                2.44                      0.40   
3                         0.0                2.14                      1.26   
4                         0.0                2.71                      0.41   
..                        ...                 ...                       ...   
935                       0.0                1.08                      0.20   
936                       0.0               11.10                      0.80   
937                       0.0                1.35                      0.46   
938                       0.0               13.22                      0.41   
939                       0.0                1.82                      0.04   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   6.06                     0.00                 25   
1                   4.71                     0.00                 21   
2                   3.91                     0.00                 30   
3                   2.83                     0.00                 29   
4                   5.04                     0.00                 36   
..                   ...                      ...                ...   
935                 6.80                     0.00                 17   
936                 6.24                     0.05                 73   
937                 6.28                     0.00                 18   
938                 5.89                     0.00                 88   
939                 4.25                     0.00                 23   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                     13                   328               728      1985   
1                     19                   217               776      1797   
2                     11                   181              1218      1776   
3                     34                   209               726      1745   
4                     10                   221               773      1863   
..                   ...                   ...               ...       ...   
935                    4                   245              1174      2847   
936                   19                   217              1131      3710   
937                   11                   224              1187      2832   
938                   12                   213              1127      3832   
939                    1                   137               770      1849   

    day_of_week  
0       Tuesday  
1     Wednesday  
2      Thursday  
3        Friday  
4      Saturday  
..          ...  
935      Sunday  
936      Monday  
937     Tuesday  
938   Wednesday  
939    Thursday  

[940 rows x 16 columns]

In [24]:
dfS[['Date', 'Time']] = dfS['SleepDay'].str.split(' ', 1, expand=True)

/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/1015236849.py:1: FutureWarning: In a future version of pandas all arguments of StringMethods.split except for the argument 'pat' will be keyword-only.
  dfS[['Date', 'Time']] = dfS['SleepDay'].str.split(' ', 1, expand=True)
/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/1015236849.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfS[['Date', 'Time']] = dfS['SleepDay'].str.split(' ', 1, expand=True)
/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/1015236849.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation:

In [25]:
dfS['Date'] = pd.to_datetime(dfS['Date'], format='%m/%d/%Y')
dfS['day_of_week'] = dfS['Date'].dt.day_name()

/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/1353775894.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfS['Date'] = pd.to_datetime(dfS['Date'], format='%m/%d/%Y')
/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/1353775894.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfS['day_of_week'] = dfS['Date'].dt.day_name()


In [26]:
dfHStep[['Date', 'Time', 'AM/PM']] = dfHStep['ActivityHour'].str.split(' ', 2, expand=True)

/var/folders/sv/hr00cdg153nd74z0xfz926y40000gn/T/ipykernel_41852/3653198968.py:1: FutureWarning: In a future version of pandas all arguments of StringMethods.split except for the argument 'pat' will be keyword-only.
  dfHStep[['Date', 'Time', 'AM/PM']] = dfHStep['ActivityHour'].str.split(' ', 2, expand=True)


In [27]:
dfHStep['Hour'] = dfHStep['Time'] + " " + dfHStep['AM/PM']
dfHStep

Id           ActivityHour  StepTotal       Date      Time  \
0      1503960366  4/12/2016 12:00:00 AM        373  4/12/2016  12:00:00   
1      1503960366   4/12/2016 1:00:00 AM        160  4/12/2016   1:00:00   
2      1503960366   4/12/2016 2:00:00 AM        151  4/12/2016   2:00:00   
3      1503960366   4/12/2016 3:00:00 AM          0  4/12/2016   3:00:00   
4      1503960366   4/12/2016 4:00:00 AM          0  4/12/2016   4:00:00   
...           ...                    ...        ...        ...       ...   
22094  8877689391  5/12/2016 10:00:00 AM        514  5/12/2016  10:00:00   
22095  8877689391  5/12/2016 11:00:00 AM       1407  5/12/2016  11:00:00   
22096  8877689391  5/12/2016 12:00:00 PM       3135  5/12/2016  12:00:00   
22097  8877689391   5/12/2016 1:00:00 PM        307  5/12/2016   1:00:00   
22098  8877689391   5/12/2016 2:00:00 PM        457  5/12/2016   2:00:00   

      AM/PM         Hour  
0        AM  12:00:00 AM  
1        AM   1:00:00 AM  
2        AM   2:00:00 AM  
3        AM   3:00:00 AM  
4        AM   4:00:00 AM  
...     ...          ...  
22094    AM  10:00:00 AM  
22095    AM  11:00:00 AM  
22096    PM  12:00:00 PM  
22097    PM   1:00:00 PM  
22098    PM   2:00:00 PM  

[22099 rows x 7 columns]

In [28]:
dfHStep['Date'] = pd.to_datetime(dfHStep['Date'], format='%m/%d/%Y')
dfHStep['day_of_week'] = dfHStep['Date'].dt.day_name()

## Data Merging

In [29]:
dfActS = pd.merge(dfAct, dfS,  how='left', left_on=['Id','ActivityDate'], right_on = ['Id','Date'])

In [30]:
dfActS

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366   2016-04-12       13162       8.500000         8.500000   
1    1503960366   2016-04-13       10735       6.970000         6.970000   
2    1503960366   2016-04-14       10460       6.740000         6.740000   
3    1503960366   2016-04-15        9762       6.280000         6.280000   
4    1503960366   2016-04-16       12669       8.160000         8.160000   
..          ...          ...         ...            ...              ...   
935  8877689391   2016-05-08       10686       8.110000         8.110000   
936  8877689391   2016-05-09       20226      18.250000        18.250000   
937  8877689391   2016-05-10       10733       8.150000         8.150000   
938  8877689391   2016-05-11       21420      19.559999        19.559999   
939  8877689391   2016-05-12        8064       6.120000         6.120000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                1.88                      0.55   
1                         0.0                1.57                      0.69   
2                         0.0                2.44                      0.40   
3                         0.0                2.14                      1.26   
4                         0.0                2.71                      0.41   
..                        ...                 ...                       ...   
935                       0.0                1.08                      0.20   
936                       0.0               11.10                      0.80   
937                       0.0                1.35                      0.46   
938                       0.0               13.22                      0.41   
939                       0.0                1.82                      0.04   

     LightActiveDistance  SedentaryActiveDistance  ...  SedentaryMinutes  \
0                   6.06                     0.00  ...               728   
1                   4.71                     0.00  ...               776   
2                   3.91                     0.00  ...              1218   
3                   2.83                     0.00  ...               726   
4                   5.04                     0.00  ...               773   
..                   ...                      ...  ...               ...   
935                 6.80                     0.00  ...              1174   
936                 6.24                     0.05  ...              1131   
937                 6.28                     0.00  ...              1187   
938                 5.89                     0.00  ...              1127   
939                 4.25                     0.00  ...               770   

     Calories  day_of_week_x               SleepDay  TotalSleepRecords  \
0        1985        Tuesday  4/12/2016 12:00:00 AM                1.0   
1        1797      Wednesday  4/13/2016 12:00:00 AM                2.0   
2        1776       Thursday                    NaN                NaN   
3        1745         Friday  4/15/2016 12:00:00 AM                1.0   
4        1863       Saturday  4/16/2016 12:00:00 AM                2.0   
..        ...            ...                    ...                ...   
935      2847         Sunday                    NaN                NaN   
936      3710         Monday                    NaN                NaN   
937      2832        Tuesday                    NaN                NaN   
938      3832      Wednesday                    NaN                NaN   
939      1849       Thursday                    NaN                NaN   

    TotalMinutesAsleep TotalTimeInBed       Date         Time  day_of_week_y  
0                327.0          346.0 2016-04-12  12:00:00 AM        Tuesday  
1                384.0          407.0 2016-04-13  12:00:00 AM      Wednesday  
2                  NaN            NaN        NaT          NaN            NaN  
3                412.0          4

In [31]:
dfActS.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed  
count   940.000000         410.000000          410.000000      410.000000  
mean   2303.609574           1.119512          419.173171      458.482927  
std     718.166862           0.346636          118.635918      127.455140  
min       0.000000           1.000000           58.000000       61.000000  
25%    1828.500000           1.000000          361.000000      403.750000  
50%    2134.000000           1.000000          432.500000      463.000000  
75%    2793.250000           1.000000          490.000000      526.000000  
max    4900.000000           3.000000          796.000000      961.000000

### Key Takeaways:
    * Average sleep hours is 7 Hours.

    

## Exporting the dataframes for visualisations

In [32]:
dfActS.to_csv('DailyActivitySleep.csv', index=False)

In [33]:
dfS.to_csv('DailySleep.csv', index=False)

In [34]:
dfHStep.to_csv('HourlyStep.csv', index=False)